### PR TITLE
chore: improve pnpm usage

### PR DESCRIPTION
**Pull request by SuperchupuDev**
## Summary
- Removes pnpm lockfiles from some package directories. In pnpm monorepos there should only be the root one
- Bumps all js dependencies (including major updates, hopefully nothing breaks)
- Sorts `.gitattributes`
- Adds `shared` to `pnpm-workspace.yaml`
- Adds the `packageManager` field for corepack users. For more info see https://nodejs.org/api/corepack.html. It makes the pnpm workflow use that field. Also removes `engines.pnpm`
- Minor package.json improvements

For maintainers:
- **Consider using pnpm via corepack, that way every single maintainer can stay in sync with the same pnpm version specified in the `packageManager` field**
- **Bumping deps should be done at the root directory with the `-r` (recursive) flag to avoid future monorepo problems**

## Test Plan

CI should pass

### CODE REVIEW

**Comment by Conaclos** [comment]
Do you think it could be worth adding some instructions in our CONTRIBUTING guide?

**Comment by SuperchupuDev** [comment]
Sounds good, will do

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -240,19 +240,19 @@ The npm module `packages/@biomejs/biome` contains Biome's Node.js API that suppo
 
 For testing and developing, you need to build these packages, following the steps:
 
-1. install [wasm-pack](https://rustwasm.github.io/wasm-pack/installer/) globally;
-2. run the `build` command inside the package `backend-jsonrpc`;
-3. run the `build` and `build:wasm-node-dev` commands inside the package `js-api` (folder `packages/@biomejs/js-api`);
-4. run `pnpm i` inside the package `js-api` (folder `packages/@biomejs/js-api`), this will link the WebAssembly bindings and the
-   JSON-RPC bindings;
+1. install pnpm via [corepack](https://nodejs.org/api/corepack.html) by running `corepack enable`
```
Comment (SuperchupuDev) [inline CONTRIBUTING.md:243]:
```suggestion
1. install pnpm via [corepack](https://nodejs.org/api/corepack.html) by running `corepack enable`;
```

File: .gitattributes [hunk 2]
```diff
@@ -1,25 +1,25 @@
 * text=auto eol=lf
```
Comment (ematipico) [inline .gitattributes:1]: How are these ordered now? It's adding a comment

**Review by ematipico — APPROVED** [review]
Thank you!

File: .gitattributes (hunk 2 above)
Comment (SuperchupuDev) [reply .gitattributes:1]: each type of files gets sorted alphabetically, i.e all generated files alphabetically then all special syntax files alphabetically (special syntax being when the language of a file is specified). should i add a comment?

**Review by Conaclos — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
 	"keywords": [],
 	"author": "",
 	"devDependencies": {
-		"shared": "file:./shared",
-		"@changesets/cli": "^2.26.2"
-	}
+		"@biomejs/shared": "workspace:^"
+	},
+	"packageManager": "pnpm@8.15.4"
```
Comment (anonrig) [inline package.json:15]: Is there a reason why we don't append the shasum to this value? By default core pack should have added it?

**Review by anonrig — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (SuperchupuDev) [reply package.json:15]: i'm used to removing it in my small personal projects and also saw some big repos that don't use it, but yeah its a good idea to have it, will add

**Comment by SuperchupuDev** [comment]
i think my last commit fixes netlify not using cache
image

previous discussion:
image

**Comment by SuperchupuDev** [comment]
ci is throwing errors about no space left on device on the rust setup actions. my last few commits don't add anything that could cause that problem

**Comment by ematipico** [comment]
> ci is throwing errors about no space left on device on the rust setup actions. my last few commits don't add anything that could cause that problem

Yeah it's unrelated to your changes, feel free to merge your PR

**Pull request merged**